nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/queue/GetAzureQueueStorage.java

```diff
@@ -0,0 +1,208 @@
+        @WritesAttribute(attribute = "azure.queue.expirationTime", description = "The time when the message will expire from the queue storage"),
+        @WritesAttribute(attribute = "azure.queue.messageId", description = "The ID of the retrieved message"),
+        @WritesAttribute(attribute = "azure.queue.popReceipt", description = "The pop receipt of the retrieved message"),
+})
+public class GetAzureQueueStorage extends AbstractAzureQueueStorage {
+
+    public static final PropertyDescriptor AUTO_DELETE = new PropertyDescriptor.Builder()
+            .name("auto-delete-messages")
+            .displayName("Auto Delete Messages")
+            .description("Specifies whether the received message is to be automatically deleted from the queue.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("batch-size")
+            .displayName("Batch Size")
+            .description("The number of messages to be retrieved from the queue.")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("32")
+            .build();
+
+    public static final PropertyDescriptor VISIBILITY_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("visibility-timeout")
+            .displayName("Visibility Timeout")
+            .description("The duration during which the retrieved message should be invisible to other consumers.")
+            .required(true)
+            .defaultValue("30 secs")
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    private static final List<PropertyDescriptor> properties = Collections.unmodifiableList(Arrays.asList(
+            AzureStorageUtils.ACCOUNT_NAME, AzureStorageUtils.ACCOUNT_KEY, AzureStorageUtils.PROP_SAS_TOKEN, QUEUE, AUTO_DELETE,
+            BATCH_SIZE, VISIBILITY_TIMEOUT));
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return Collections.singleton(REL_SUCCESS);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        final int visibilityTimeoutInSecs = context.getProperty(VISIBILITY_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
+        final int batchSize = context.getProperty(BATCH_SIZE).asInteger();
+        final boolean autoDelete = context.getProperty(AUTO_DELETE).asBoolean();
+
+        final Iterable<CloudQueueMessage> retrievedMessagesIterable;
+
+        try {
+            retrievedMessagesIterable = cloudQueue.retrieveMessages(batchSize, visibilityTimeoutInSecs, null, null);
```

Comment:
how does this work in a clustered approach? should this processor only run on primary node? if yes, could be indicated in the doc/processor description.

Reply:
Checked it and in clustered scenarios it works as expected. Only one node at a time retrieves the message. Tested with varying message counts as well different number of nodes.